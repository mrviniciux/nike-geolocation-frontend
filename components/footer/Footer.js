import { Row, Col } from 'antd';
import styles from './Footer.module.scss'

const Footer = () => (
      <Row className={styles['footer-container']}>
        <Col span={6} className={styles['main-links-container']}>
            <p className={styles['main-links']}><a href="#">Encontre uma loja da nike</a></p>
            <p className={styles['main-links']}><a href="#">Cadastre-se para receber novidades</a></p>
            <p className={styles['main-links']}><a href="#">Mapa do site</a></p>
        </Col>
        <Col span={6} className={styles['caption-links-container']}>
            <h2>Ajuda</h2>
            <p className={styles['caption-link']}><a href="#">Dúvidas Gerais</a></p>
            <p className={styles['caption-link']}><a href="#">Encontre seu Tamanho</a></p>
            <p className={styles['caption-link']}><a href="#">Entregas</a></p>
            <p className={styles['caption-link']}><a href="#">Pedidos</a></p>
            <p className={styles['caption-link']}><a href="#">Trocas e Devoluções</a></p>
            <p className={styles['caption-link']}><a href="#">Editar consentimento</a></p>
            <p className={styles['caption-link']}><a href="#">Preferências de Cookie</a></p>
            <p className={styles['caption-link']}><a href="#">Pagamentos</a></p>
            <p className={styles['caption-link']}><a href="#">Produtos</a></p>
            <p className={styles['caption-link']}><a href="#">Corporativo</a></p>
            <p className={styles['caption-link']}><a href="#">Fale Conosco</a></p>
        </Col>
        <Col span={6} className={styles['about-nike']}>
            <h2>Sobre a nike</h2>
            <p className={styles['caption-link']}><a href="#">Feito para Jogar</a></p>
            <p className={styles['caption-link']}><a href="#">Sustentabilidade</a></p>
        </Col>
        <Col span={6} className={styles['social-payment']}>
            <h2>Redes sociais</h2>
            <img src="/assets/logo_redes_sociais_facebook.svg" />
            <img src="/assets/logo_redes_sociais_instagram.svg" />
            <img src="/assets/logo_redes_sociais_youtube.svg" />

            <h2>Formas de pagamento</h2>
            <img src="/assets/pgto_Mastercard.svg"></img>
            <img src="/assets/pgto_Visa.svg"></img>
            <img src="/assets/pgto_Amex.svg"></img>
            <img src="/assets/pgto_Elo.svg"></img>
            <img src="/assets/pgto_Hipercard.svg"></img>
            <img src="/assets/pgto_Discover.svg"></img>
            <img src="/assets/pgto_Boleto.svg"></img>
        </Col>
      </Row>
);

export default Footer;
