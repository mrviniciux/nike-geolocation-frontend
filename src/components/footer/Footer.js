import { Row, Col } from 'antd';
import styles from './Footer.module.scss';

const Footer = () => (
  <footer>
    <Row className={styles['footer-container']}>
      <Col
        data-testid="footer-main-links"
        span={6}
        className={styles['main-links-container']}
      >
        <p className={styles['main-links']}>
          <a href="#">Encontre uma loja da nike</a>
        </p>
        <p className={styles['main-links']}>
          <a href="#">Cadastre-se para receber novidades</a>
        </p>
        <p className={styles['main-links']}>
          <a href="#">Mapa do site</a>
        </p>
      </Col>
      <Col
        data-testid="footer-caption-links"
        span={6}
        className={styles['caption-links-container']}
      >
        <h2>Ajuda</h2>
        <p className={styles['caption-link']}>
          <a href="#">Dúvidas Gerais</a>
        </p>
        <p className={styles['caption-link']}>
          <a href="#">Encontre seu Tamanho</a>
        </p>
        <p className={styles['caption-link']}>
          <a href="#">Entregas</a>
        </p>
        <p className={styles['caption-link']}>
          <a href="#">Pedidos</a>
        </p>
        <p className={styles['caption-link']}>
          <a href="#">Trocas e Devoluções</a>
        </p>
        <p className={styles['caption-link']}>
          <a href="#">Editar consentimento</a>
        </p>
        <p className={styles['caption-link']}>
          <a href="#">Preferências de Cookie</a>
        </p>
        <p className={styles['caption-link']}>
          <a href="#">Pagamentos</a>
        </p>
        <p className={styles['caption-link']}>
          <a href="#">Produtos</a>
        </p>
        <p className={styles['caption-link']}>
          <a href="#">Corporativo</a>
        </p>
        <p className={styles['caption-link']}>
          <a href="#">Fale Conosco</a>
        </p>
      </Col>
      <Col
        data-testid="footer-about-nike-links"
        span={6}
        className={styles['about-nike']}
      >
        <h2>Sobre a nike</h2>
        <p className={styles['caption-link']}>
          <a href="#">Feito para Jogar</a>
        </p>
        <p className={styles['caption-link']}>
          <a href="#">Sustentabilidade</a>
        </p>
      </Col>
      <Col span={6} className={styles['social-payment']}>
        <h2>Redes sociais</h2>
        <img
          data-testid="social-media"
          src="/assets/logo_redes_sociais_facebook.svg"
          alt="facebook logo"
        />
        <img
          data-testid="social-media"
          src="/assets/logo_redes_sociais_instagram.svg"
          alt="instagram logo"
        />
        <img
          data-testid="social-media"
          src="/assets/logo_redes_sociais_youtube.svg"
          alt="youtube logo"
        />

        <h2>Formas de pagamento</h2>
        <img
          data-testid="payment-option"
          src="/assets/pgto_Mastercard.svg"
          alt="visa logo"
        />
        <img
          data-testid="payment-option"
          src="/assets/pgto_Visa.svg"
          alt="visa logo"
        />
        <img
          data-testid="payment-option"
          src="/assets/pgto_Amex.svg"
          alt="amex logo"
        />
        <img
          data-testid="payment-option"
          src="/assets/pgto_Elo.svg"
          alt="elo logo"
        />
        <img
          data-testid="payment-option"
          src="/assets/pgto_Hipercard.svg"
          alt="hipercard logo"
        />
        <img
          data-testid="payment-option"
          src="/assets/pgto_Discover.svg"
          alt="discover logo"
        />
        <img
          data-testid="payment-option"
          src="/assets/pgto_Boleto.svg"
          alt="boleto logo"
        />
      </Col>
    </Row>
    <Row className={styles['copyright-container']}>
      <Col span={12}>
        <ul className={styles['menu-copyright']}>
          <li>
            <a href="#">Brasil</a>
          </li>
          <li>
            <a href="#">Politica de privacidade</a>
          </li>
          <li>
            <a href="#">Termos de uso</a>
          </li>
        </ul>
      </Col>
      <Col className={styles['copyright-text-container']} span={12}>
        <span className={styles['copyright-text']}>
          © 2021 Nike. Todos os direitos reservados. Fisia Comércio de Produtos
          Esportivos Ltda - CNPJ: 59.546.515/0016-10 Rua José Semião Rodrigues
          Agostinho, 1370 - Galpão Unidade 17 - Nível 0, 1 e 2 - CEP 06833-300 -
          Embu das Artes/SP.
        </span>
      </Col>
    </Row>
  </footer>
);

export default Footer;
