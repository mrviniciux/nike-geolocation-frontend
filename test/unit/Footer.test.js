/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import '../matchMedia.mock.js';
import React from 'react';
import Footer from '../../src/components/footer/Footer';
import { render } from '@testing-library/react';

describe('Footer rendering tests', () => {
  it('should have main links', () => {
    const { getByTestId } = render(<Footer />);

    expect(getByTestId('footer-main-links')).toBeInTheDocument();
  });

  it('should have caption links', () => {
    const { getByTestId } = render(<Footer />);

    expect(getByTestId('footer-caption-links')).toBeInTheDocument();
  });

  it('should have about nike links', () => {
    const { getByTestId } = render(<Footer />);

    expect(getByTestId('footer-about-nike-links')).toBeInTheDocument();
  });

  it('must have all payment options as described', () => {
    const { getAllByTestId } = render(<Footer />);

    expect(getAllByTestId('payment-option')).toHaveLength(6);
  });

  it('must have all social media icons as described', () => {
    const { getAllByTestId } = render(<Footer />);

    expect(getAllByTestId('social-media')).toHaveLength(3);
  });
});
