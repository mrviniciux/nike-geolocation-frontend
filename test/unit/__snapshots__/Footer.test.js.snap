// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer rendering tests/snapshot should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer>
        <div
          class="ant-row"
          role="row"
        >
          <div
            class="ant-col ant-col-6"
            role="cell"
          >
            <p>
              <a
                href="#"
              >
                Encontre uma loja da nike
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Cadastre-se para receber novidades
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Mapa do site
              </a>
            </p>
          </div>
          <div
            class="ant-col ant-col-6"
            role="cell"
          >
            <h2>
              Ajuda
            </h2>
            <p>
              <a
                href="#"
              >
                Dúvidas Gerais
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Encontre seu Tamanho
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Entregas
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Pedidos
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Trocas e Devoluções
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Editar consentimento
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Preferências de Cookie
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Pagamentos
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Produtos
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Corporativo
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Fale Conosco
              </a>
            </p>
          </div>
          <div
            class="ant-col ant-col-6"
            role="cell"
          >
            <h2>
              Sobre a nike
            </h2>
            <p>
              <a
                href="#"
              >
                Feito para Jogar
              </a>
            </p>
            <p>
              <a
                href="#"
              >
                Sustentabilidade
              </a>
            </p>
          </div>
          <div
            class="ant-col ant-col-6"
            role="cell"
          >
            <h2>
              Redes sociais
            </h2>
            <img
              src="/assets/logo_redes_sociais_facebook.svg"
            />
            <img
              src="/assets/logo_redes_sociais_instagram.svg"
            />
            <img
              src="/assets/logo_redes_sociais_youtube.svg"
            />
            <h2>
              Formas de pagamento
            </h2>
            <img
              src="/assets/pgto_Mastercard.svg"
            />
            <img
              src="/assets/pgto_Visa.svg"
            />
            <img
              src="/assets/pgto_Amex.svg"
            />
            <img
              src="/assets/pgto_Elo.svg"
            />
            <img
              src="/assets/pgto_Hipercard.svg"
            />
            <img
              src="/assets/pgto_Discover.svg"
            />
            <img
              src="/assets/pgto_Boleto.svg"
            />
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer>
      <div
        class="ant-row"
        role="row"
      >
        <div
          class="ant-col ant-col-6"
          role="cell"
        >
          <p>
            <a
              href="#"
            >
              Encontre uma loja da nike
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Cadastre-se para receber novidades
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Mapa do site
            </a>
          </p>
        </div>
        <div
          class="ant-col ant-col-6"
          role="cell"
        >
          <h2>
            Ajuda
          </h2>
          <p>
            <a
              href="#"
            >
              Dúvidas Gerais
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Encontre seu Tamanho
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Entregas
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Pedidos
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Trocas e Devoluções
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Editar consentimento
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Preferências de Cookie
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Pagamentos
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Produtos
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Corporativo
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Fale Conosco
            </a>
          </p>
        </div>
        <div
          class="ant-col ant-col-6"
          role="cell"
        >
          <h2>
            Sobre a nike
          </h2>
          <p>
            <a
              href="#"
            >
              Feito para Jogar
            </a>
          </p>
          <p>
            <a
              href="#"
            >
              Sustentabilidade
            </a>
          </p>
        </div>
        <div
          class="ant-col ant-col-6"
          role="cell"
        >
          <h2>
            Redes sociais
          </h2>
          <img
            src="/assets/logo_redes_sociais_facebook.svg"
          />
          <img
            src="/assets/logo_redes_sociais_instagram.svg"
          />
          <img
            src="/assets/logo_redes_sociais_youtube.svg"
          />
          <h2>
            Formas de pagamento
          </h2>
          <img
            src="/assets/pgto_Mastercard.svg"
          />
          <img
            src="/assets/pgto_Visa.svg"
          />
          <img
            src="/assets/pgto_Amex.svg"
          />
          <img
            src="/assets/pgto_Elo.svg"
          />
          <img
            src="/assets/pgto_Hipercard.svg"
          />
          <img
            src="/assets/pgto_Discover.svg"
          />
          <img
            src="/assets/pgto_Boleto.svg"
          />
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
